iosApp/iosApp/ContentView.swift

```diff
@@ -97,16 +97,19 @@ struct ContentView: View {
     @ViewBuilder
     var nearbyTab: some View {
         VStack {
-            if contentVM.searchEnabled, nearbyVM.navigationStack.lastSafe() == .nearby {
-                TextField("Find nearby transit", text: $searchObserver.searchText)
-                SearchView(
-                    query: searchObserver.debouncedText,
-                    nearbyVM: nearbyVM,
-                    searchVM: searchVM
-                )
-            }
             locationAuthHeader
-            mapWithSheets
+            ZStack(alignment: .top) {
+                mapWithSheets
+                VStack(alignment: .trailing, spacing: 0) {
+                    if contentVM.searchEnabled, nearbyVM.navigationStack.lastSafe() == .nearby {
+                        SearchOverlay(searchObserver: searchObserver, nearbyVM: nearbyVM, searchVM: searchVM)
+                    }
+                    if !searchObserver.isSearching, !viewportProvider.viewport.isFollowing,
+                       locationDataManager.currentLocation != nil {
+                        RecenterButton { Task { viewportProvider.follow() } }
+                    }
```

Comment:
question: Could this be kept within the HomeMapView if there is a new prop like `hideRecenter`? I think it would be nice to keep the recenter button localized with the rest of the map if possible

Reply:
No, I don't think so. It needs to be in this VStack so that it will layout properly below the search bar when it exists, or in the top corner when search is gone. I think it makes sense separate since it is a map overlay. If we have other buttons or need to clean this up in the future, I think having a general map overlay view which includes search and any buttons could make sense.